src/pdm/cli/commands/add.py

```diff
@@ -174,7 +174,7 @@ def do_add(
         if sync:
             do_sync(
                 project,
-                selection=GroupSelection(project, groups=[group], default=False),
+                selection=GroupSelection(project, groups=[group], dev=selection.dev, default=False),
```

Comment:
Oops, that fails a test case, when the group is a dev group but `--dev` is not passed explicitly. I guess it's not the correct way to fix, neither for `pdm remove`.

I think you should fix the `_translated_groups()` function, so that when dev groups don't exist in the lock file they won't be added to the group_set

Reply:
Gotcha, I will take a look later.

Reply:
You can use a set intersection to exclude the non-exist groups